domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/InterestDate.java

```diff
@@ -30,10 +38,13 @@
 
     private final String reason;
 
-    public InterestDate(InterestDateType type, LocalDate date, String reason) {
+    private final InterestEndDateType endDate;
+
+    public InterestDate(InterestDateType type, LocalDate date, String reason, InterestEndDateType endDate) {
         this.type = type;
         this.date = date;
         this.reason = reason;
+        this.endDate = endDate == null ? InterestEndDateType.SUBMISSION : endDate;
```

Comment:
me thinks this end date has the wrong default,
This would mean interest would stop accumulating at submission, whereas currently its at judgment